include/BlitzHelpers.hpp

```diff
@@ -190,4 +318,116 @@ namespace blitzdg {
 
         return C;
     }
+
+    template <typename T>
+    std::pair<std::vector<index_type>, std::vector<index_type>>
+    unique(const matrix_type<T>& A) {
+        using details::CompareExact;
+
+        std::vector<index_type> gather(A.rows()), scatter(A.rows());
+        std::iota(gather.begin(), gather.end(), index_type(0));
+        CompareExact<T> comp(A);
+        std::sort(gather.begin(), gather.end(), comp);
+        index_type i = 0;
+        for (index_type k = 0; k < A.rows(); ++k) {
+            auto curr = gather[i];
+            auto next = gather[k];
+            if (comp(curr, next) || comp(next, curr)) { // A(next,:) != A(curr,:)
+                ++i;
+                gather[i] = next;
+            }
+            scatter[next] = i;
+        }
+        gather.resize(i + 1);
+        return std::make_pair(std::move(gather), std::move(scatter));
+    }
+
+    template <typename T>
+    std::pair<std::vector<index_type>, std::vector<index_type>>
+    uniquetol(const matrix_type<T>& A, real_type tol = real_type(0)) {
+        using details::CompareEQ;
+        using details::CompareEQByDim;
+        using details::CompareExact;
+        using details::getOrdering;
+
+        if (tol <= real_type(0)) {
+            return unique(A);
+        }
+        std::vector<index_type> gather(A.rows()), scatter(A.rows());
+        std::iota(gather.begin(), gather.end(), index_type(0));
+        auto order = getOrdering(A);
```

Comment:
I was lazy with my naming here. I would change `getOrdering` to `orderColsByVariance`.